apps/studio/components/interfaces/QueryPerformance/QueryPerformanceGrid.tsx

```diff
@@ -168,7 +168,21 @@ export const QueryPerformanceGrid = ({ queryPerformanceQuery }: QueryPerformance
     return result
   })
 
-  const reportData = data ?? []
+  const reportData = useMemo(() => {
+    const allData = data ?? []
+
+    if (!search || search.trim() === '') {
+      return allData
+    }
+
+    const searchTerm = search.toLowerCase()
+    return allData.filter((row: any) => {
+      // Search in the query text
+      const queryText = row.query?.toLowerCase() || ''
+      return queryText.includes(searchTerm)
```

Comment:
hmm is there any context as to why we're moving away from server side searching to client side?

Reply:
JFYI as well - with this we're currently actually doing _both_ server and client side searching ([ref](https://github.com/supabase/supabase/blob/feat/query-perf-simplify-filters/apps/studio/pages/project/%5Bref%5D/advisors/query-performance.tsx#L46)) as the search query is also being passed into the RQ hook that retrieves the report data

Reply:
side thought - im thinking this could get confusing from a UX POV (although i might be over worrying too)

sorting is done on the server side, but searching is done on the client side (based on the current data that's shown) - could this be confusing? wdyt?

Reply:
I think we're back to that with the latest changes. Server side search, last 20 results.